### PR TITLE
[CLEANUP Beta] Cleanup Ember.get

### LINKED ISSUE #11733: Cleanup Ember.get
- [ ] Remove support for globals: `Ember.get('App.foo')` and `Ember.get(null, 'App.foo')`
- [ ] Remove support for `this`: `Ember.get(object, 'this.foo')`
- [ ] Enforce strict usage with two arguments: `Ember.get(object, path)`
  - assert object is a non-null object & path is a string.

### PR DESCRIPTION
**Pull request by ghost**
Resolves #11733

### CODE REVIEW

File: packages/ember-metal/lib/property_get.js [hunk 1]
```diff
@@ -51,6 +51,8 @@ export let UNHANDLED_GET = symbol('UNHANDLED_GET');
   @public
 */
 export function get(obj, keyName) {
+  Ember.assert(`Get must be called with two arguments; an object and a property key`, arguments.length >= 2);
+
```
Comment (martndemus) [inline packages/ember-metal/lib/property_get.js]: It seems that `_getPath` ([here](https://github.com/martndemus/ember.js/blob/cleanup-ember-get/packages/ember-metal/lib/property_get.js#L174)), calls get with three arguments, can anyone explain me why?

Comment (mmun) [inline packages/ember-metal/lib/property_get.js]: Looks like it should not.

**Comment by martndemus** [comment]
There is an [error in the test suite](https://github.com/emberjs/ember.js/blob/9df93dcdc094beedc336c9e3a71da7f9c10b5049/packages/ember-runtime/tests/computed/reduce_computed_macros_test.js#L1067)

File: packages/ember-metal/lib/property_get.js [hunk 2]
```diff
@@ -51,24 +50,17 @@ export let UNHANDLED_GET = symbol('UNHANDLED_GET');
   @public
 */
 export function get(obj, keyName) {
+  Ember.assert(`Get must be called with two arguments; an object and a property key`, arguments.length === 2);
+  Ember.assert(`Cannot call get with '${keyName}' on an undefined object.`, obj !== undefined && obj !== null);
+  Ember.assert(`Cannot call get with ${keyName} key.`, typeof keyName === 'string');
```
Comment (cibernox) [inline packages/ember-metal/lib/property_get.js]: This assertion could be clearer with something like `The key provided to get must be a string, you passed ${keyName}`

Comment (martndemus) [inline packages/ember-metal/lib/property_get.js]: I just took the existing message. But if that's better I could change it.

Comment (mixonic) [inline packages/ember-metal/lib/property_get.js]: Seems like a good improvement

**Comment by mixonic** [comment]
This is sweet :clap:

**Comment by martndemus** [comment]
Test suite fails on `Ember.view` using global lookup for the `emptyView` property.
I have created PR #11754 for this.

**Comment by stefanpenner** [comment]
Lgtm

**Comment by mixonic** [comment]
@rwjblue please suggest squash or merge if you are happy as is.

**Comment by martndemus** [comment]
I squashed all but one commit.
That commit might belong in another PR as it's more of an bug fix I think.

**Pull request merged**